src/loss.cpp

```diff
@@ -0,0 +1,81 @@
+#include <vector>
+
+#include <Eigen/Core>
+#include <deepworks/loss.hpp>
+
+
+using ConstMatrix = Eigen::Map<const Eigen::Matrix<float, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>;
+using Matrix = Eigen::Map<Eigen::Matrix<float, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>;
+
+using ConstVector = Eigen::Map<const Eigen::Matrix<float, 1, Eigen::Dynamic, Eigen::RowMajor>>;
+using Vector = Eigen::Map<Eigen::Matrix<float, 1, Eigen::Dynamic, Eigen::RowMajor>>;
+
+
+namespace deepworks {
+namespace losses {
+
+std::vector<int> GetTargetIndices(const ConstVector target, int rows, int cols) {
+    std::vector<int> slice(rows);
+    for (int i = 0; i < rows; ++i) {
+        slice[i] = static_cast<int>(target(0, i)) + cols * i;
+    }
+    return slice;
+}
+
+float CPUNLLLoss(Matrix X, const ConstVector target) {
+    int rows = X.rows();
+    int cols = X.cols();
```

Comment:
Can make it more simply for understanding and rename it to batch_size and n_classes?